authority/config.go

```diff
@@ -0,0 +1,233 @@
+package authority
+
+import (
+	"crypto/x509"
+	"encoding/json"
+	"io/ioutil"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/juju/errors"
+)
+
+const (
+	// UserNoticeQualifierType defines id-qt-unotice
+	UserNoticeQualifierType = "id-qt-unotice"
+	// CpsQualifierType defines id-qt-cps
+	CpsQualifierType = "id-qt-cps"
+)
+
+// AllowedCSRFields provides booleans for fields in the CSR.
+// If a AllowedCSRFields is not present in a CertProfile,
+// all of these fields may be copied from the CSR into the signed certificate.
+// If a AllowedCSRFields *is* present in a CertProfile,
+// only those fields with a `true` value in the AllowedCSRFields may
+// be copied from the CSR to the signed certificate.
+// Note that some of these fields, like Subject, can be provided or
+// partially provided through the API.
+// Since API clients are expected to be trusted, but CSRs are not, fields
+// provided through the API are not subject to validation through this
+// mechanism.
+type AllowedCSRFields struct {
```

Comment:
Where is this used?

Reply:
When cert request is received, the engine will check which fields in CSR to allow.
If the client specifies `email` but it's not allowed by the cert profile, then it will not be used as provided in CSR